solidity/contracts/BancorNetwork.sol

```diff
@@ -122,35 +131,24 @@ contract BancorNetwork is IBancorNetwork, TokenHolder, ContractRegistryClient, F
         // verify that the account which should receive the conversion result is whitelisted
         require(isWhitelisted(_path, _for));
 
-        // handle msg.value
-        if (etherTokens[_path[0]]) {
-            require(msg.value == _amount);
-            if (!isOwnerAnETHConverter(_path[1]))
-                IEtherToken(_path[0]).deposit.value(msg.value)();
+        // validate msg.value and prepare the source tokens for the conversion
+        handleSourceTokens(_path[0], ISmartToken(_path[1]), _amount);
+
+        bool affiliateFeeEnabled = false;
+        if (address(_affiliateAccount) == 0) {
+            require(_affiliateFee == 0);
         }
         else {
-            require(msg.value == 0);
+            require(0 < _affiliateFee && _affiliateFee <= maxAffiliateFee);
+            affiliateFeeEnabled = true;
         }
 
         // convert and get the resulting amount
-        uint256 amount = convertByPath(_path, _amount, _minReturn, _affiliateAccount, _affiliateFee);
-
-        // finished the conversion, transfer the funds to the target account
-        // if the target token is an ether token, withdraw the tokens and send them as ETH
-        // otherwise, transfer the tokens as is
-        IERC20Token toToken = _path[_path.length - 1];
-        if (etherTokens[toToken]) {
-            if (isOwnerAnETHConverter(_path[_path.length - 2]))
-                _for.transfer(amount);
-            else
-                IEtherToken(toToken).withdrawTo(_for, amount);
-        }
-        else {
-            uint256 prevBalance = toToken.balanceOf(_for);
-            INonStandardERC20(toToken).transfer(_for, amount);
-            uint256 postBalance = toToken.balanceOf(_for);
-            require(postBalance > prevBalance);
-        }
+        ConversionStep[] memory data = createConversionData(_path, _minReturn, _for, affiliateFeeEnabled);

```

Comment:
it wouldn't be correct, since conversion data can actually be composed of more stuff other than the steps, right now it's just the steps but might change in the future.
in general the argument is the concept of all the data needed for conversion, whatever it may be.

Reply:
In that case, the current struct name is not correct.
Either both reflect conversion step, or both reflect something more general.
It doesn't make sense to use different names IMO.
The variable name should somehow imply about its type here IMO.

Reply:
I don't see why - it's abstraction vs. implementation - the abstract name is 'data' and the implementation is whatever we see as the current requirements are.
I don't have a problem with changing the type if we decide to add more stuff to the data and create a new structure, I just want to make sure it involves minimal code without adding unnecessary overhead now and the current implementation accomplishes both.